### PR TITLE
[cleanup] Fix docs after TransferObject change

### PR DESCRIPTION
**Pull request by tnowacki**
- Fix json_rpc doc
- Fix transaction doc

### CODE REVIEW

File: doc/src/build/transactions.md [hunk 1]
```diff
@@ -37,14 +37,24 @@ In addition to the common metadata above, a publish transaction includes Package
...
+
+This transaction type is similar to the Transfer Object transaction type, but the input object type must be a Sui coin--that is, an object of type `sui::coin::Coin<sui::sui::SUI>`. The benefit of this transaction type is that a separate coin object is not needed for gas. The gas payment is taken from the Sui coin being transferred.
+Optionally, an amount can be specified for partial transfers.
+
+In addition to the common metadata above, a transfer Sui transaction includes the following fields:
+
+* Input: An object reference pointing to a `sui::coin::Coin<sui::sui::SUI>` object owned by the sender.
+* (Optional) Amount: An unsigned integer encoding the amount that the recipient will receive. The amount will be debited from the input object, wrapped in a freshly created coin object, and sent to the corresponding recipient address. The value of the input object must be greater than or equal to the amount specified.
 
 ### Join
```
Comment (tnowacki) [inline doc/src/build/transactions.md:59]: @lxfind is this section outdated?

File: doc/src/build/transactions.md [hunk 2]
```diff
@@ -37,14 +37,24 @@ In addition to the common metadata above, a publish transaction includes Package
...
-* Input: An object reference pointing to a mutable object owned by the sender. The object must be of type `sui::coin::Coin<T>` with arbitrary `T`--that is, any fungible token. The gas input object from above cannot also appear as an object input.
-* Recipients: The addresses that will receive payments from this transfer. This list must be non-empty.
-* Amounts: A list of unsigned integers encoding the amount that each recipient will receive. This list must be the same length as the recipients list. Each amount will be debited from the input object, wrapped in a freshly created coin object, and sent to the corresponding recipient address. The value of the input object must be greater than or equal to the sum of the amounts.
+In addition to the common metadata above, a transfer object transaction includes the following fields:
+
+* Input: An object reference pointing to a mutable object owned by the sender. The object must be of type that allows for public transfers--that is, any type with the `store` ability. The gas input object from above cannot also appear as the object input.
+* Recipients: The address that will receive payments from this transfer.
+
+### Transfer Sui
```
Comment (tnowacki) [inline doc/src/build/transactions.md:49]: Wrote a new section on Transfer Sui

File: doc/src/build/transactions.md [hunk 3]
```diff
@@ -37,14 +37,24 @@ In addition to the common metadata above, a publish transaction includes Package
...
-* Input: An object reference pointing to a mutable object owned by the sender. The object must be of type `sui::coin::Coin<T>` with arbitrary `T`--that is, any fungible token. The gas input object from above cannot also appear as an object input.
-* Recipients: The addresses that will receive payments from this transfer. This list must be non-empty.
-* Amounts: A list of unsigned integers encoding the amount that each recipient will receive. This list must be the same length as the recipients list. Each amount will be debited from the input object, wrapped in a freshly created coin object, and sent to the corresponding recipient address. The value of the input object must be greater than or equal to the sum of the amounts.
+In addition to the common metadata above, a transfer object transaction includes the following fields:
+
+* Input: An object reference pointing to a mutable object owned by the sender. The object must be of type that allows for public transfers--that is, any type with the `store` ability. The gas input object from above cannot also appear as the object input.
+* Recipients: The address that will receive payments from this transfer.
```
Comment (Clay-Mysten) [inline doc/src/build/transactions.md:47]: Is this really plural, Recipients? If so, are multiple addresses supported?

If the latter is also correct, I would address by making this:
"Recipients: The address(es)..."

Comment (tnowacki) [reply doc/src/build/transactions.md:47]: Ah should be a single address I think.

I think this file in general was really out of date (and still is)

File: doc/src/build/transactions.md [hunk 4]
```diff
@@ -37,14 +37,24 @@ In addition to the common metadata above, a publish transaction includes Package
...
+
+### Transfer Sui
+
+This transaction type is similar to the Transfer Object transaction type, but the input object type must be a Sui coin--that is, an object of type `sui::coin::Coin<sui::sui::SUI>`. The benefit of this transaction type is that a separate coin object is not needed for gas. The gas payment is taken from the Sui coin being transferred.
```
Comment (Clay-Mysten) [inline doc/src/build/transactions.md:51]: I think we should capitalize SUI here when referencing coins/tokens per:
https://docs.sui.io/learn/about-sui#what-sui-is

@econmysten, is that your intent with the disambiguation you and I discussed earlier?

File: doc/src/build/transactions.md [hunk 5]
```diff
@@ -37,14 +37,24 @@ In addition to the common metadata above, a publish transaction includes Package
...
+Optionally, an amount can be specified for partial transfers.
+
+In addition to the common metadata above, a transfer Sui transaction includes the following fields:
```
Comment (Clay-Mysten) [inline doc/src/build/transactions.md:54]: Same comment: I believe this should be "transfer SUI transaction" yet defer to you and Alonso.

**Review by Clay-Mysten — APPROVED** [review]
Thanks so much, Todd! I created a second commit to fix the little capitalization issues I found and added Alonso to confirm I did this as expected. Thanks again!

File: doc/src/build/transactions.md (hunk 3 above)
Comment (Clay-Mysten) [reply doc/src/build/transactions.md:47]: > Ah should be a single address I think.
> 
> I think this file in general was really out of date (and still is)

Thanks! Do you know who would be the best person to review this admittedly old overview of the transaction process?

File: doc/src/build/transactions.md (hunk 4 above)
Comment (Clay-Mysten) [reply doc/src/build/transactions.md]: I tried to address this in a commit here.

**Comment by tnowacki** [comment]
> Thanks so much, Todd! I created a second commit to fix the little capitalization issues I found and added Alonso to confirm I did this as expected. Thanks again!

Ah shit, I think I accidentally just blew those changes away on my push. Sorry!

File: doc/src/build/transactions.md (hunk 3 above)
Comment (tnowacki) [reply doc/src/build/transactions.md]: @lxfind ?

**Comment by Clay-Mysten** [comment]
> > Thanks so much, Todd! I created a second commit to fix the little capitalization issues I found and added Alonso to confirm I did this as expected. Thanks again!
> 
> Ah shit, I think I accidentally just blew those changes away on my push. Sorry!

No worries! I will check/fix right now.

File: doc/src/build/transactions.md (hunk 4 above)
Comment (tnowacki) [reply doc/src/build/transactions.md]: I think and blew up those changes, and hopefully have them back now

Comment (Clay-Mysten) [reply doc/src/build/transactions.md]: Thanks! Sorry for not coordinating with you. :-(

**Comment by Clay-Mysten** [comment]
> > Thanks so much, Todd! I created a second commit to fix the little capitalization issues I found and added Alonso to confirm I did this as expected. Thanks again!
> 
> Ah shit, I think I accidentally just blew those changes away on my push. Sorry!

Sorry about that. Your changes LGTM.

**Review by Clay-Mysten — APPROVED** [review]
Thanks, Todd! All LGTM now.

**Pull request merged**